### PR TITLE
PAL: Resolve libc-location build-time on FreeBSD.

### PR DESCRIPTION
**Pull request by josteink**
corerun fails to initialize properly because it can't load `libc.so`. This patch resolves this at build-time, and results in corerun being able to execute a Hello World sample application.

**Read that as: corerun is live on FreeBSD!**

This closes https://github.com/dotnet/coreclr/issues/801

@janvorli Is this an OK solution by your books?

### CODE REVIEW

File: src/pal/src/configure.cmake [hunk 1]
```diff
@@ -96,6 +96,7 @@ check_cxx_symbol_exists(_DEBUG sys/user.h USER_H_DEFINES_DEBUG)
 check_cxx_symbol_exists(_SC_PHYS_PAGES unistd.h HAVE__SC_PHYS_PAGES)
 check_cxx_symbol_exists(_SC_AVPHYS_PAGES unistd.h HAVE__SC_AVPHYS_PAGES)
 
+
```
Comment (kangaroo) [inline src/pal/src/configure.cmake]: remove whitespace.

File: src/pal/src/loader/module.cpp [hunk 2]
```diff
@@ -1349,8 +1349,8 @@ static HMODULE LOADLoadLibrary(LPCSTR ShortAsciiName, BOOL fDynamic)
     {
 #if defined(__APPLE__)
         ShortAsciiName = "libc.dylib";
-#elif defined(__FreeBSD__)
-        ShortAsciiName = "libc.so";
+#elif defined(__FreeBSD__) && defined(FREEBSD_LIBC)
```
Comment (kangaroo) [inline src/pal/src/loader/module.cpp]: If we want to check that FREEBSD_LIBC is defined, it shouldn't fall down to the Linux case, it should error.

Comment (josteink) [inline src/pal/src/loader/module.cpp]: Fair enough.

**Comment by janvorli** [comment]
@josteink Can you also check if there is something like LIBC_SO defined in the FreeBSD headers? It would be nice if we could use that instead. If not, let's go with this solution.

**Comment by josteink** [comment]
@janvorli I grepped over entire `/usr/include` without finding anything. I'd love to have something less ... redundant, but I'm not sure how that can be done.

**Comment by janvorli** [comment]
So you don't have any lib-names.h file? On Linux, it resides in
/usr/include/x86_64-linux-gnu/gnu/lib-names.h

**Comment by josteink** [comment]
Sadly not it seems.

**Comment by janhenke** [comment]
That CI failure is caused by a problem in the CI infrastructure.

**Comment by kangaroo** [comment]
@dotnet-bot test this please

**Comment by kangaroo** [comment]
LGTM

**Comment by ajensenwaud** [comment]
Well done Jostein!

**Pull request merged**